lib/jwt/jwa/unsupported.rb

```diff
@@ -3,16 +3,16 @@
 module JWT
   module JWA
     module Unsupported
-      module_function
+      class << self
+        include JWT::JWA::SigningAlgorithm
 
-      SUPPORTED = [].freeze
+        def sign(*)
+          raise NotImplementedError, 'Unsupported signing method'
```

Comment:
Lets not do `NotImplementedError` as it's not inherited from StandardError it can cause strangeness..